File: 2019-01-02/10828/10828.c
#include <stdio.h>
#include <string.h>

#define PUSH "push"
#define POP "pop"
#define SIZE "size"
#define EMPTY "empty"
#define TOP "top"

int arr[10005] = {0,};

void push(int number);
int pop();
int size();
int empty();
int top();
void printResult(int n);
 
int main(N){
	char method[10];
	int input;

	scanf("%d", &N);

	for(int i = 0; i < N; i++){
		scanf("%s", method);
		if(!strcmp(method, PUSH)){
			scanf("%d", &input);
			push(input);
		}
		else if(!strcmp(method, POP))
			printResult(pop());

		else if(!strcmp(method, SIZE))
			printResult(size());
		
		else if(!strcmp(method, EMPTY))
			printResult(empty());
		
		else if(!strcmp(method, TOP))
			printResult(top());
	}
	return 0;
}

int ix = -1; 

void printResult(int n){
	printf("%d\n", n);
}

void push(int number){
	ix++;
	arr[ix] = number;
}

int pop(){
	int value;
	if(ix < 0){
		value = -1;
	}else{
		value = arr[ix];
		ix--;
	}

	return value;
}

int size(){
	return ix+1;
}

int empty(){
	if(ix < 0)
		return 1;
	return 0;
}

int top(){
	if(ix < 0)
		return -1;
	return arr[ix];
}

File: 2019-01-02/10828/10828.java
import java.util.*;
import java.io.*;

class Main{
	static Stack<Integer> stack = new Stack<Integer>(); 

	final static String PUSH = "push";
	final static String POP = "pop";
	final static String EMPTY = "empty";
	final static String TOP = "top";
	final static String SIZE = "size";

	public static void main(String[] args) throws Exception{
		BufferedReader bfr = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(bfr.readLine());

		for(int i = 0; i < N; i++){
			String[] command = bfr.readLine().split(" ");

			if(command[0].equals(PUSH))
				push(Integer.parseInt(command[1]));

			else if(command[0].equals(POP) || command[0].equals(TOP)){
				if(isEmpty())
					printResult(-1);
				else{
					printResult(peek());
					if(command[0].equals(POP))
						pop();
				}
			}

			else if(command[0].equals(EMPTY)){
				if(isEmpty())
					printResult(1);
				else
					printResult(0);
			}

			else if(command[0].equals(SIZE))
				printResult(size());
			
			
		}
	}

	private static int peek(){
		return stack.peek();
	}

	private static Boolean isEmpty(){
		return stack.empty();
	}

	private static int getSize(){
		return stack.size();
	}

	private static void printResult(int num){
		System.out.println(num);
	}

	private static void push(int num){
		stack.push(num);
	}

	private static int pop(){
		stack.pop();
	}
}